src/Scripting/Core/Hosting/Resolvers/RuntimeMetadataReferenceResolver.cs

```diff
@@ -165,7 +165,7 @@ public override ImmutableArray<PortableExecutableReference> ResolveReference(str
 
                 if (PathResolver != null)
                 {
-                    string resolvedPath = PathResolver.ResolvePath(reference, baseFilePath);
+                    string resolvedPath = PathResolver.ResolvePath(StandardFileSystem.Instance, reference, baseFilePath);
```

Comment:
>StandardFileSystem.Instance, [](start = 67, length = 28)

This is adding dependency on file system where it wasn't before. `PathResolver` provided abstraction for FileExists. This code didn't need to care about the specific implementation.

Reply:
The resolver already had a depedency on the file system, it just preferred inheritance as the means to mock it out. That really complicates the job of anyone who is trying to write a test about it. This makes it explicit that the file system is involved here.

Reply:
This method does not have dependency on the file system. The dependency is encapsulated in RelativePathResolver, which is a parameter passed in the constructor.